### PR TITLE
add topic filter persistence and API

### PR DESCRIPTION
**Pull request by DC2-DanielKrueger**
https://hivemq.kanbanize.com/ctrl_board/57/cards/25322/details/

### CODE REVIEW

File: hivemq-edge/src/main/java/com/hivemq/persistence/topicfilter/TopicFilterPersistenceReaderWriter.java [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+        } catch (final JsonProcessingException e) {
+            log.error(
+                    "Error while trying to persist the topic filters on disc. Exception happened during serialization of topic filters:",
+                    e);
+            throw new RuntimeException(e);
+        } catch (final IOException e) {
+            log.error("Error while trying to persist the topic filters on disc. Exception happened during writing of " +
+                    PERSISTENCE_FILE_NAME +
+                    ":", e);
+            throw new RuntimeException(e);
```
Comment (codepitbull) [inline hivemq-edge/src/main/java/com/hivemq/persistence/topicfilter/TopicFilterPersistenceReaderWriter.java:86]: Why not use UnrecoverableException here?

File: hivemq-edge/src/main/java/com/hivemq/persistence/topicfilter/TopicFilterPersistenceReaderWriter.java [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+
+        try {
+            final TopicFilterPersistenceEntity persistenceEntity = convertToEntity(tags);
+            final String xml = xmlMapper.writeValueAsString(persistenceEntity);
+            Files.writeString(persistenceFile.toPath(), xml);
+        } catch (final JsonProcessingException e) {
+            log.error(
+                    "Error while trying to persist the topic filters on disc. Exception happened during serialization of topic filters:",
+                    e);
+            throw new RuntimeException(e);
```
Comment (codepitbull) [inline hivemq-edge/src/main/java/com/hivemq/persistence/topicfilter/TopicFilterPersistenceReaderWriter.java:81]: Why not use UnrecoverableException here?

**Review by codepitbull — APPROVED** [review]
Just a question about exceptions left.
Up to you to decide if it should be adopted :)

File: hivemq-edge/src/main/java/com/hivemq/persistence/topicfilter/TopicFilterPersistenceReaderWriter.java (hunk 1 above)
Comment (DC2-DanielKrueger) [reply hivemq-edge/src/main/java/com/hivemq/persistence/topicfilter/TopicFilterPersistenceReaderWriter.java:86]: Wont do something outside of startup

File: hivemq-edge/src/main/java/com/hivemq/persistence/topicfilter/TopicFilterPersistenceReaderWriter.java (hunk 2 above)
Comment (DC2-DanielKrueger) [reply hivemq-edge/src/main/java/com/hivemq/persistence/topicfilter/TopicFilterPersistenceReaderWriter.java:81]: Wont do something outside of startup

**Pull request merged**